chandra_aca/tests/test_aca_image.py

```diff
@@ -482,3 +489,110 @@ def test_flicker_test_sequence():
             ]
         ),
     )
+
+
+def images_check_range(start, stop, img_table, *, bgsub):
+    tstart = CxoTime(start).secs
+    tstop = CxoTime(stop).secs
+
+    # Check that start and stop of the range are within 4.1 secs of the requested times
+    assert np.all(np.abs(img_table["TIME"][0] - tstart) < 4.1)
+    assert np.all(np.abs(img_table["TIME"][-1] - tstop) < 4.1)
+
+    # Check that all the times in the table are within the requested range
+    assert np.all((img_table["TIME"] >= tstart) & (img_table["TIME"] < tstop))
+
+    # Check that all the slots are in there
+    assert np.all(np.isin(np.arange(8), img_table["IMGNUM"]))
+
+    # Check that if the table has BGSUB column that IMG - DARK = BGSUB
+    if bgsub:
+        assert np.allclose(
+            img_table["IMG"] - img_table["IMG_DARK"], img_table["IMG_BGSUB"]
+        )
+
+    # If there's a DARK column, then check it against the dark image from the dark cal
+    if bgsub:
+        dc = get_dark_cal_props(
+            tstart, select="nearest", include_image=True, aca_image=True
+        )
+        full_img_dark = dc["image"]
+        tccd_dark = dc["ccd_temp"]
+        for row in img_table:
+            dark_row = row["IMG_DARK"]
+            tccd_row = row["T_CCD_SMOOTH"]
+            row8x8 = row["IMGROW0_8X8"]
+            col8x8 = row["IMGCOL0_8X8"]
+
+            # If row8x8 or col8x8 is < 0 just skip the test for now
+            if row8x8 < 0 or col8x8 < 0:
```

Comment:
And I just cheated a bit here as I didn't want to deal with negative indexes in the test.